keras_cv/visualization/plot_image_gallery.py

```diff
@@ -92,27 +91,69 @@ def unpackage_tfds_inputs(inputs):
             "to be true."
         )
 
-    fig = plt.figure(figsize=(cols * scale, rows * scale))
-    fig.tight_layout()  # Or equivalently,  "plt.tight_layout()"
-    plt.subplots_adjust(wspace=0, hspace=0)
-    plt.margins(x=0, y=0)
-    plt.axis("off")
+    if isinstance(images, tf.data.Dataset):
+
+        # Find final dataset batch size
+        sample = next(iter(images.take(1)))
+        if len(sample[image_key].shape) == 3:
```

Comment:
If `rows` and `cols` are `None`, let's assume images are batched and throw otherwise.

If they are specified, we should assume that the dataset is unbatched and take the first `rows * cols` images.

(This way we can also drop the `batch_size` parameter)

Reply:
Got it. Changed